okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -972,6 +975,44 @@ static HttpUrl getChecked(String url) throws MalformedURLException, UnknownHostE
     return PublicSuffixDatabase.get().getEffectiveTldPlusOne(host);
   }
 
+  public static String displayHost(Builder.DisplayMode displayMode, String host) {
+    switch (displayMode) {
+      case SAFE:
+        String unicodeHost = IDN.toUnicode(host, 0);
+        if (unicodeHost.equals(host)) {
+          return host;
+        }
+        return isDisplaySafe(unicodeHost) ? unicodeHost : host;
+      case UNICODE:
+        return IDN.toUnicode(host, 0);
+      default:
+        return host;
+    }
+  }
+
+  /**
+   * Returns the unicode host if the host is considered IDN safe.
```

Comment:
ALSO – I believe in Chrome the display URLs take a Locale, and only use Unicode when the locales match. Should this do likewise?